### PR TITLE
initial function to run soil model

### PR DESCRIPTION
**Pull request by Clare2D**
Lot's of to dos still on this - so don't merge yet, but perhaps you can download the file and work on it. 

Some of the todos:
- call the variables directly from parameter files
- create new parameter files for bare profile and carbon inputs
- clean this up - should the loop happen in this function? 
- Perhaps remove sensitivity analysis
- add new functions for results (possibly elsewhere)

### CODE REVIEW

**Comment by JeremieO** [comment]
@Clare2D  I added my scripts. Can you review that quickly ? Let me know if I need to change something, and feel free to merge whenever you want.

File: scripts/calc_functions_soil_modelling.R [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  animals = merge(x = filter(animal_inputs,scenario==scenario_chosen & parcel_ID==parcel), y = animal_factors, by = "species", all.x = TRUE) %>% 
+    mutate (C_inputs_manure_kg_per_ha_per_year= n_animals*c_kg_per_year_per_animal/area*grazing_days/365)
+  tC_inputs_per_ha_per_year = sum(animals$C_inputs_manure_kg_per_ha_per_year)*1e-3
+  return(tC_inputs_per_ha_per_year)
+}
+
+
+get_monthly_Cinputs_agroforestry <- function (agroforestry_inputs, agroforestry_factors, scenario_chosen, parcel){
+  zone=ifelse(lat_farmer<57,"Temperate","Boreal")
+  trees = merge(x = filter(agroforestry_inputs,scenario==scenario_chosen & parcel_ID==parcel), y = filter(agroforestry_factors,climatic_zone==zone), by = "tree_species", all.x = TRUE) %>%
```
Comment (Clare2D) [inline scripts/calc_functions_soil_modelling.R:13]: I think some cleaning of this at some stage might be good - or perhaps just some additional line breaks to make it simpler to see everything.

File: scripts/run_soil_model.R [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+                                                      scenario=scenario,
+                                                      agroforestry_Cinput=get_monthly_Cinputs_agroforestry(agroforestry_inputs, agroforestry_factors, scenario, parcel),
+                                                      animal_Cinput=get_monthly_Cinputs_animals(animal_inputs, animal_factors, scenario, parcel),
+                                                      crop_Cinputs=get_monthly_Cinputs_crop(crop_inputs, crop_data, scenario, parcel),
+                                                      pasture_Cinputs=get_monthly_Cinputs_pasture(pasture_inputs, pasture_data, scenario, parcel)))
+    }
+  }
+  parcels_Cinput <- parcels_Cinput %>% mutate(tot_Cinputs=agroforestry_Cinput+animal_Cinput+crop_Cinputs+pasture_Cinputs)
+  
+  write.csv(parcels_Cinput,file.path(project_loc,project_name,"results/parcels_Cinput.csv"), row.names = TRUE)
```
Comment (Clare2D) [inline scripts/run_soil_model.R:52]: here better to change it to ...."results", "parcels_Cinput.csv") 
This means that there is never an issue for anyone with the /

File: scripts/run_soil_model.R [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+                                   soil_thick = mean_input$soil_thick[1],
+                                   clay = mean_input$clay[1],
+                                   pE = mean_input$pE[1],
+                                   PS = starting_soil_content,
+                                   tilling_factor = mean_input$tilling_factor[1])
+    print(tail(C0_df,1))
+    new_starting_soil_content <- as.numeric(tail(C0_df,1))[c(1:5)]
+    #new_starting_soil_content <- estimate_starting_soil_content(SOC=18.1,clay=mean_input$clay[1])
+    
+    mean=c(list(c(Cinput,rep(NA,11))),list(c(1.44,rep(NA,11))),list(c(12/12,rep(NA,11))),list(weather_data$future_temperature),list(weather_data$future_precipitation*365*30.4*24*3600),list(weather_data$future_pevap*365*30.4*24*3600),list(c(30,rep(NA,11))),list(c(15,rep(NA,11))),list(c(0.75,rep(NA,11))),list(c(1.0,rep(NA,11))))
```
Comment (Clare2D) [inline scripts/run_soil_model.R:82]: I would add in some line breaks here as well!

File: scripts/run_soil_model.R [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+    step_holistic <- diff(temp_holistic$yearly_mean)
+    step_in_table <- rbind(step_in_table,data.frame(parcel_ID=parcel,
+                                                    year=year_temp,
+                                                    baseline=step_baseline,
+                                                    holistic=step_holistic))
+  }
+  
+}
+
+# #### SENSITIVITY ANALYSIS
```
Comment (Clare2D) [inline scripts/run_soil_model.R:180]: remove commented sections?

File: scripts/run_soil_model_Fransisco.R [hunk 5]
```diff
@@ -0,0 +1,28 @@
+#######################################################################
```
Comment (Clare2D) [inline scripts/run_soil_model_Fransisco.R:1]: Can you remove this file? I don't think this one needs to go public

File: scripts/weather_data_pulling_functions.R [hunk 6]
```diff
@@ -0,0 +1,131 @@
+# Weather functions
```
Comment (Clare2D) [inline scripts/weather_data_pulling_functions.R:1]: This file to modelling-data :)

File: scripts/weather_data_pulling_functions.R [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+  # Spatial resolution is 1 x 1 km. 
+  # Temporal resolution was quarter to year, averaged over 20 years.
+  # We assumed that January to March is the wettest quarter, July to September is the driest quarter 
+  # while April to June and Octover to December corresponds to the annual mean.
+  #######################
+  
+  # set ncpath and 
+  ncdir <- "sis-biodiversity-cmip5-regional_data/sis-biodiversity-cmip5-regional-potential_evapotranspiration"
+  ncname <- "downscaled_monthly_pevaporation_1x1km.nc"
+  ncpath <- file.path(weatherDB_loc, ncdir, ncname)
```
Comment (Clare2D) [inline scripts/weather_data_pulling_functions.R:101]: this is looking good!

**Review by Clare2D — COMMENTED** [review]
Almost there. The cleaning could happen later, but mainly a few extra line breaks where things extend over a screen width. And checking the weather data to modelling-data and removing the Francisco file and it looks good!

**Pull request merged**